src/modal/modal-container.component.ts

```diff
@@ -76,12 +77,18 @@ export class ModalContainerComponent implements OnInit, OnDestroy {
     }
   }
 
-  @HostListener('click', ['$event'])
-  onClick(event: MouseEvent): void {
+  @HostListener('mousedown', ['$event'])
+  onClickStarted(event: MouseEvent): void {
+    this.clickStartedInside = event.target !== this._element.nativeElement;
+  }
+
+  @HostListener('mouseup', ['$event'])
+  onClickStop(event: MouseEvent): void {
+    const clickedInside = this.clickStartedInside || event.target !== this._element.nativeElement;
     if (
       this.config.ignoreBackdropClick ||
       this.config.backdrop === 'static' ||
-      event.target !== this._element.nativeElement
+      clickedInside
     ) {
       return;
     }
```

Comment:
You could consider to reset`clickStartedInside` to default value after it was used for condition to not use a previous result of onClickStart if there was not mousedown inside but rather only mouseup.